matlab_simulation/05-estimation/kalman_filter/kf_example2.m

```diff
@@ -0,0 +1,109 @@
+% Kalman Filter Example 2: 2D Omnidirectional AUV example
+% 'Estimation 1' topic slide 77
+% State : position and velocity in north, east directions
+% Motion : motion in north and east directions
+% Measurement : measure position in north and east directions
+% Controller : Thrust in north and east directions
+
+clear all
+clc
+close all
+
+% Example number
+example = 2;
+
+% Create AVI object
+makemovie = 1;
+if(makemovie)
+    vidObj = VideoWriter('example2kf.avi');
+    vidObj.Quality = 100;
+    vidObj.FrameRate = 8;
+    open(vidObj);
+end
+
+% Discrete time step
+dt = 0.1;
+
+% Prior
+mu = zeros(4,1); % mean (mu)
+S = 1*eye(4);% covariance (Sigma)
+% S = 0.1*eye(4);% covariance (Sigma)
+% S = 0.01*eye(4);% covariance (Sigma)
+
+% Get motion model and measurement model
+[A,B,R,n] = motion_model(example,dt);
+[C,D,Q,m] = measurement_model(example);
+
+[QRE, QRe] = eig(Q.QR);
+[QE, Qe] = eig(Q.Q);
+[RE, Re] = eig (R);
+
+% Store in a structure
+ssm.A = A;
+ssm.B = B;
+ssm.C = C;
+ssm.D = D;
+ssm.R = R;
+ssm.Q = Q;
+ssm.n = n;
+ssm.m = m;
+
+% Simulation Initializations
+Tf = 10;
+T = 0:dt:Tf;
+u = 10*[sin(2*T);cos(T)];
+n = length(A(1,:));
+x = zeros(n,length(T));
+x(:,1) = zeros(n,1);
+m = length(C(:,1));
+y = zeros(m,length(T));
+mup_S = zeros(n,length(T));
+mu_S = zeros(n,length(T));
+
+%% Main loop
+for t=2:length(T)
+    %% Simulation
+    % Select a motion disturbance
+    e = RE*sqrt(Re)*randn(n,1);
+    % Update state
+    x(:,t) = A*x(:,t-1)+ B*u(:,t) + e;
+
+    % Take measurement
+    % Select a motion disturbance
+    d = QRE*sqrt(QRe)*randn(m,1);
+    % Determine measurement
+    y(:,t) = C*x(:,t) + d;
+
+    
+    %% Kalman Filter Estimation
+    [mu,S,mup,Sp,K] = kalman_filter(ssm,mu,S,u(:,t),y(:,t),example,t);
+    
+    % Store results
+    mup_S(:,t) = mup;
+    mu_S(:,t) = mu;
+    K_S(:,t) = [K(:,1); K(:,2)];
+  
+    %% Plot results
```

Comment:
Maybe put an x and y label on this plot. You can add the labels after the for loop so that it doesn't slow it down.